packages/teams-js/src/private/copilot.ts

```diff
@@ -41,12 +48,53 @@ export namespace copilot {
      *
      * @throws Error if {@linkcode app.initialize} has not successfully completed
      */
-    export function getEligibilityInfo(): AppEligibilityInformation {
+    export function getEligibilityInfo(): Promise<AppEligibilityInformation> {
       ensureInitialized(runtime);
       if (!isSupported()) {
         throw errorNotSupportedOnPlatform;
       }
-      return runtime.hostVersionsInfo!.appEligibilityInformation!;
+
+      // Return the eligibility information if it is already available
+      if (runtime.hostVersionsInfo?.appEligibilityInformation) {
+        return Promise.resolve(runtime.hostVersionsInfo!.appEligibilityInformation);
+      }
+
+      // Send message to host SDK to get eligibility information
+      return sendMessageToParentAsync<AppEligibilityInformation | SdkError>(
+        getApiVersionTag(copilotTelemetryVersionNumber, ApiName.Copilot_Eligibility_GetEligibilityInfo),
+        'copilot.eligibility.getEligibilityInfo',
+      ).then((response: AppEligibilityInformation | SdkError) => {
+        if (isSdkError(response)) {
```

Comment:
In case, some error comes that is not of type sdkError, this function cast it as `AppEligibilityInformation` and send to the app. I think if we have a type-guard for the response as well that would be good